src/js/schemas/service-schema/HealthChecks.js

```diff
@@ -0,0 +1,132 @@
+import FormUtil from '../../utils/FormUtil';
+
+let HealthChecks = {
+  type: 'object',
+  title: 'Health Checks',
+  properties: {
+    healthChecks: {
+      type: 'array',
+      duplicable: true,
+      addLabel: 'Add Another Health Check',
+      getter: function (service) {
+        let healthChecks = service.getHealthChecks();
+        if (healthChecks == null) {
+          return [];
+        }
+
+        return healthChecks;
+      },
+      deleteButtonTop: true,
+      filterProperties: function (service = {}, instanceDefinition) {
+        let properties = HealthChecks
+          .properties
+          .healthChecks
+          .itemShape
+          .properties;
+
+        instanceDefinition.forEach(function (definition) {
+          let prop = definition.name;
+          if (FormUtil.isFieldInstanceOfProp('healthChecks', definition)) {
+            prop = FormUtil.getPropKey(definition.name);
+          }
+          if (properties[prop].shouldShow) {
+            definition.formElementClass = {
+              'hidden-form-element': !properties[prop].shouldShow(service)
```

Comment:
Yeah I chose this path because it had the least resistance. Since the compiled definition is passed around everywhere, it's nice to just mutate the props on it instead of slicing and splicing. I agree that we should have a cleanup in one of the Utils. I'll put up another PR for that.